src/sentry/search/snuba/executors.py

```diff
@@ -465,34 +475,112 @@ def better_priority_aggregation(
     end: datetime,
     aggregate_kwargs: PrioritySortWeights,
 ) -> Sequence[str]:
-    issue_age_weight = 1  # [0, 5]
-    event_age_weight = 1  # [0, 5]
+    issue_age_weight = 1  # [1, 5]
+    event_age_weight = 1  # [1, 5]
     log_level_weight = aggregate_kwargs["log_level"]  # [0, 10]
     stacktrace_weight = aggregate_kwargs["has_stacktrace"]  # [0, 3]
     max_pow = 16
     min_score = 0.01
     event_age_hours = "divide(now() - timestamp, 3600)"
-    event_halflife_hours = aggregate_kwargs["event_halflife_hours"]  # halves score every 4 hours
+    event_halflife_hours = aggregate_kwargs["event_halflife_hours"]  # halves score every x hours
     issue_age_hours = "divide(now() - min(timestamp), 3600)"
-    issue_halflife_hours = 24 * 7  # issues half in value every week
-
-    # TODO: we have no way of aggregating to account for frequency:
-    #  (errors within group) / (total events grouped by project)
-    # frequency_weight = aggregate_kwargs["frequency"]  # [0, 5]
-    # frequency_score = f"countIf(and(greater(toDateTime('{start_str}'), timestamp), lessOrEquals(toDateTime('{end_str}'), timestamp))) / count()"
 
     log_level_score = "multiIf(equals(level, 'fatal'), 1.0, equals(level, 'error'), 0.66, equals(level, 'warning'), 0.33, 0.0)"
     stacktrace_score = "if(notEmpty(exception_stacks.type), 1.0, 0.0)"
+    # event_agg_rank:
+    #   ls = log_level_score    {1.0, 0.66, 0.33, 0}
+    #   lw = log_level_weight   [0, 10]
+    #   ss = stacktrace_score   {1.0, 0.0}
+    #   sw = stacktrace_weight  [0, 3]
+    #   as = event_age_score    [1, 0]
+    #   aw = event_age_weight   [1, 5]
+    #
+    #        (ls * lw) + (ss * sw) + (as * aw)     min(f(x)  = 0, when individual scores are all 0
+    # f(x) = ---------------------------------  ,  max(f(x)) = 1, when individual scores are all 1
+    #                  lw + sw + aw
+    #
     event_agg_numerator = f"plus(plus(multiply({log_level_score}, {log_level_weight}), multiply({stacktrace_score}, {stacktrace_weight})), {event_age_weight})"
     event_agg_denominator = (
         f"plus(plus({log_level_weight}, {stacktrace_weight}), {event_age_weight})"
     )
-    event_agg_rank = f"divide({event_agg_numerator}, {event_agg_denominator})"
+    event_agg_rank = f"divide({event_agg_numerator}, {event_agg_denominator})"  # values from [0, 1]
+
+    v2 = aggregate_kwargs["v2"]
+
+    if not v2:
+        issue_halflife_hours = 24 * 7  # issues half in value every week
+        aggregate_event_score = f"greatest({min_score}, sum(divide({event_agg_rank}, pow(2, least({max_pow}, divide({event_age_hours}, {event_halflife_hours}))))))"
+        aggregate_issue_score = f"greatest({min_score}, divide({issue_age_weight}, pow(2, least({max_pow}, divide({issue_age_hours}, {issue_halflife_hours})))))"
+        return [f"multiply({aggregate_event_score}, {aggregate_issue_score})", ""]
+    else:
+        #  * more aggressive issue half-life decay (24*7 hours -> to 4 hours)
```

Comment:
I'm fine with trying it but this is pretty aggressive. Issues that happen over a weekend could be totally missed due to this

Reply:
Good point, I followed up with @trillville in the notion doc to see why we're being so aggressive here.